transport.go

```diff
@@ -41,3 +46,22 @@ func (m *Manager) Register(s *grpc.Server) {
 func (m *Manager) Transport() raft.Transport {
 	return raftAPI{m}
 }
+
+func (m *Manager) Close() error {
+	m.connectionsMtx.Lock()
+	defer m.connectionsMtx.Unlock()
+
+	err := errCloseErr
+	for _, conn := range m.connections {
+		closeErr := conn.clientConn.Close()
```

Comment:
Before using conn.clientConn you need to temporarily pick up conn.mtx. This mutex is held by https://github.com/Jille/raft-grpc-transport/blob/master/raftapi.go#L41 until the Dial completes. The mutex should be dropped immediately after acquisition, it's just used to wait for Dial() to complete.

I'm slightly uncomfortable picking up a slow mutex (conn.mtx) while holding a fast mutex (m.connectionsMtx), but judging from the current code it should be safe.

Reply:
Good note. I added the mutex check.